### PR TITLE
NonZero operation: uncomment tests since they can be passed now

### PR DESCRIPTION
**Pull request by avoskoboinyk-lohika**
### Details:
Remove some NonZero tests from exclusions list since it supports dynamic output shape

### CODE REVIEW

**Comment by yuxu42** [comment]
@chenhu-wang could you please take a review?

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/nonzero.hpp [hunk 1]
```diff
@@ -19,18 +19,22 @@ namespace LayerTestsDefinitions {
...
+                         virtual public ov::test::SubgraphBaseTest {
 public:
     static std::string getTestCaseName(const testing::TestParamInfo<NonZeroLayerTestParamsSet>& obj);
 
 protected:
     void SetUp() override;
+    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override;
+
+protected:
+    size_t startFrom = 0, range = 10;
```
Comment (chenhu-wang) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/nonzero.hpp:37]: minor: maybe we can set range with smaller value like 2 or 5, to have more zeros in the input tensor to have a diverse distribution. In my understanding, in real models, there should be lot of zeros in input tensor.

File: src/tests/functional/inference_engine/serialization/single_layer/nonzero.cpp [hunk 2]
```diff
@@ -11,12 +11,19 @@ using namespace LayerTestsDefinitions;
...
+    serialize();
 }
 
-std::vector<InferenceEngine::SizeVector> inputDims = {
-    {7},       {1000},         {3, 5},       {65, 33},       {33, 65},
-    {1, 1000}, {223, 217, 21}, {3, 4, 5, 1}, {3, 4, 1, 5, 1}};
+std::vector<ov::test::InputShape> inputDims = {
+    {{}, {{7}}},
+    {{}, {{1000}}},
+    {{}, {{3, 5}}},
```
Comment (chenhu-wang) [inline src/tests/functional/inference_engine/serialization/single_layer/nonzero.cpp:20]: Let's add some repeat shapes to cover run time parameters cache function, which in future maybe enable in this node.
```suggestion
    {{}, {{3, 5}}, {{2, 4}}, {{3, 5}}},
```
and if possible add some lower bound and upper bound case for dynamic shape cases, take this as example:https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/cpu/single_layer_tests/mvn.cpp#L191

Comment (avoskoboinyk-lohika) [reply src/tests/functional/inference_engine/serialization/single_layer/nonzero.cpp:20]: added repeated and lower / upper bound shapes to the cpu single_layer tests

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/nonzero.hpp (hunk 1 above)
Comment (avoskoboinyk-lohika) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/nonzero.hpp]: Done (changed to 3)

**Comment by usstq** [comment]
@avoskoboinyk-lohika I enabled auto-merge, can you fix CI problems?

File: src/tests/functional/plugin/cpu/single_layer_tests/nonzero.cpp [hunk 3]
```diff
@@ -175,6 +175,26 @@ std::vector<InputShape> inShapesDynamic = {
...
+            // dynamic shape
+            {-1, -1, -1, -1, -1},
+            { // target static shapes
+                {4, 24, 5, 6, 4},
+                {8, 32, 9, 10, 2},
+                {4, 24, 5, 6, 4},
+                {16, 48, 9, 12, 2}
+            }
+        },
+        {
```
Comment (chenhu-wang) [inline src/tests/functional/plugin/cpu/single_layer_tests/nonzero.cpp:189]: /agent/_work/1/openvino/src/tests/functional/plugin/cpu/single_layer_tests/nonzero.cpp:189:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
Done processing /agent/_work/1/openvino/src/tests/functional/plugin/cpu/single_layer_tests/nonzero.cpp

Comment (chenhu-wang) [reply src/tests/functional/plugin/cpu/single_layer_tests/nonzero.cpp:189]: This cause cpplint failure in CI.

**Pull request merged**